controllers/sync.go

```diff
@@ -83,29 +66,84 @@ func chooseUpdatingComponents(spec ytv1.YtsaurusSpec, needUpdate []components.Co
 		return nil, "All components are uptodate"
 	}
 
-	switch configuredSelector {
-	case ytv1.UpdateSelectorEverything:
-		if needFullUpdate {
+	if len(configuredSelectors) == 1 && configuredSelectors[0].Class == consts.ComponentClassEverything {
```

Comment:
Here we should check for "one of configured selectors is ComponentClassEverything", because we don't reject something like `[{Type: DataNodes, Name: somename}, {Class: Everything}]`.
But I guess all others can be thrown away in `getEffectiveSelectors` if `Everything` is present.